### ISSUE TITLE
`Unpacked` `Any` `tuple` doesn't match against `TypeVarTuple`, contrary to PEP-646

### ISSUE THREAD

**Issue by colehaus** [microsoft/pyright#3890]
I believe that `Foo(Unpack[tuple[Any, ...]])` should be an acceptable type for `Foo(Unpack[Xs])` with `TypeVarTuple("Xs")`. Pyright does not accept this kind of code.

PEP-646 has code very similar to:
```py
from typing import Any, Generic, NewType, cast
from typing_extensions import TypeVarTuple, Unpack

Shape = TypeVarTuple("Shape")
Batch = NewType("Batch", int)
Height = NewType("Height", int)
Width = NewType("Width", int)
Channels = NewType("Channels", int)

class Array(Generic[Unpack[Shape]]): ...

y: Array[Unpack[tuple[Any, ...]]] = cast(Array[Unpack[tuple[Any, ...]]], None)

def expect_variadic_array(
    x: Array[Batch, Unpack[Shape]]
) -> None: ...

expect_variadic_array(y)  # Suppose to be okay according to PEP-646 but fails to type check in pyright

def expect_precise_array(
    x: Array[Batch, Height, Width, Channels]
) -> None: ...

expect_precise_array(y)  # OK
```
In the PEP, `expect_variadic_array(y)` is supposed to type check. With the pyright CLI v1.1.269, it raises the following error:
```
error: Argument of type "Array[*tuple[Any, ...]]" cannot be assigned to parameter "x" of type "Array[Batch, *Shape@expect_variadic_array]" in function "expect_variadic_array"
    TypeVar "Shape@Array" is invariant
      TypeVarTuple cannot be bound to a tuple of unknown length (reportGeneralTypeIssues)
```

This code also type checks in pyre 0.9.15.

But the presence of a specific error message suggests that pyright has chosen to diverge from the spec? The behavior specified in the PEP seems quite useful for doing incremental upgrades to shaped arrays (e.g. using `NDArray: TypeAlias = ndarray[Unpack[tuple[Any, ...]], DType]`).

**Comment by erictraut** [comment]
Thanks for pointing this out. I implemented support for PEP 646 when the PEP was still in draft form. This is an area where the PEP was modified before it was accepted in its final form, and I missed the change. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.270, which I just published. It will also be included in a future release of pylance.
